govcd/nsxt_alb_pool_test.go

```diff
@@ -0,0 +1,276 @@
+//go:build nsxt || alb || functional || ALL
+// +build nsxt alb functional ALL
+
+package govcd
+
+import (
+	"fmt"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+
+	. "gopkg.in/check.v1"
+)
+
+func (vcd *TestVCD) Test_AlbPool(check *C) {
+	if vcd.skipAdminTests {
+		check.Skip(fmt.Sprintf(TestRequiresSysAdminPrivileges, check.TestName()))
+	}
+	skipNoNsxtAlbConfiguration(vcd, check)
+	skipOpenApiEndpointTest(vcd, check, types.OpenApiPathVersion1_0_0+types.OpenApiEndpointAlbEdgeGateway)
+
+	// Setup prerequisite components
+	controller, cloud, seGroup, edge, assignment := setupAlbPoolPrerequisites(check, vcd)
+
+	// Setup Org user and connection
+	adminOrg, err := vcd.client.GetAdminOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	orgUserVcdClient, err := newOrgUserConnection(adminOrg, "alb-pool-testing", "CHANGE-ME", vcd.config.Provider.Url, true)
+	check.Assert(err, IsNil)
+
+	// Run tests with System user
+	testMinimalPoolConfig(check, edge, vcd, vcd.client)
+	testAdvancedPoolConfig(check, edge, vcd, vcd.client)
+	testPoolWithCertNoPrivateKey(check, vcd, edge.EdgeGateway.ID, vcd.client)
+	testPoolWithCertAndPrivateKey(check, vcd, edge.EdgeGateway.ID, vcd.client)
+
+	// Run tests with Org admin user
+	testMinimalPoolConfig(check, edge, vcd, orgUserVcdClient)
+	testAdvancedPoolConfig(check, edge, vcd, orgUserVcdClient)
+	testPoolWithCertNoPrivateKey(check, vcd, edge.EdgeGateway.ID, orgUserVcdClient)
+	testPoolWithCertAndPrivateKey(check, vcd, edge.EdgeGateway.ID, orgUserVcdClient)
+
+	// teardown prerequisites
+	tearDownAlbPoolPrerequisites(check, assignment, edge, seGroup, cloud, controller)
```

Comment:
I think you need to add `defer` here to cleanup even on failures.

Reply:
I relied on the cleanup job, but here it goes